homeassistant/components/media_player/plex.py

```diff
@@ -217,7 +243,11 @@ def request_configuration(host, hass, config, add_devices_callback):
     def plex_configuration_callback(data):
         """Handle configuration changes."""
         setup_plexserver(
-            host, data.get('token'), hass, config, add_devices_callback)
+                host, data.get('token'),
+                True if data.get('has_ssl') else False,
```

Comment:
I set up data validation.

Checkboxes would have been a much better alternative, but unfortunately I don't have enough experience in Polymer and the whole HA developing ecosystem to develop that feature at its best.